tests/cmdline/commands/test_data.py

```diff
@@ -332,6 +332,62 @@ def test_bandsexport(self):
         self.assertEqual(res.exit_code, 0, 'The command did not finish correctly')
         self.assertIn(b'[1.0, 3.0]', res.stdout_bytes, 'The string [1.0, 3.0] was not found in the bands' 'export')
 
+    def test_bandsexport_single_kp(self):
+        """ test issue #2462 """
+
+        """Create bands structure object."""
+        alat = 4.  # angstrom
+        cell = [
+            [
+                alat,
+                0.,
+                0.,
+            ],
+            [
+                0.,
+                alat,
+                0.,
+            ],
+            [
+                0.,
+                0.,
+                alat,
+            ],
+        ]
+        strct = StructureData(cell=cell)
+        strct.append_atom(position=(0., 0., 0.), symbols='Fe')
+        strct.append_atom(position=(alat / 2., alat / 2., alat / 2.), symbols='O')
+        strct.store()
+
```

Comment:
this part duplicate from class function `create_structure_bands`. Since creating `BandsData` without `StructureData` as its ancestors with break `data bands list`.

Reply:
I create an issue for this https://github.com/aiidateam/aiida-core/issues/3811